### ISSUE TITLE
Typescript error: __DEV__ conflicts with React Native

### ISSUE THREAD

**Issue by szymn** [apollographql/apollo-client#9039]
**Intended outcome:**

No typescript errors

**Actual outcome:**

When the client is used with react-native, Typescript throws an error: 
```
node_modules/@apollo/client/utilities/globals/global.d.ts:2:11 - error TS2451: Cannot redeclare block-scoped variable '__DEV__'.

2     const __DEV__: boolean | undefined;
            ~~~~~~~

  node_modules/@types/react-native/index.d.ts:9904:11
    9904     const __DEV__: boolean;
                   ~~~~~~~
    '__DEV__' was also declared here.
```

**How to reproduce the issue:**

1. Create a react-native project from typescript template
2. Add `@apollo/client`
3. Make sure skipLibCheck in tsconfig is false
4. Run tsc

**Versions**

  System:
    OS: macOS 11.5.2
  Binaries:
    Node: 14.18.1 - /usr/local/opt/nvm/versions/node/v14.18.1/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.15 - /usr/local/opt/nvm/versions/node/v14.18.1/bin/npm
  Browsers:
    Chrome: 95.0.4638.69
    Firefox: 87.0
    Safari: 14.1.2
  npmPackages:
    @apollo/client: 3.4.16 => 3.4.16
    apollo-link-timeout: 4.0.0 => 4.0.0

react-native: 0.66.2
typescript: 4.4.3

**Comment by danilobuerger** [comment]
Experiencing the same. Current workaround is using patch package:
```diff
diff --git a/node_modules/@apollo/client/utilities/globals/global.d.ts b/node_modules/@apollo/client/utilities/globals/global.d.ts
deleted file mode 100644
index a4d791b..0000000
--- a/node_modules/@apollo/client/utilities/globals/global.d.ts
+++ /dev/null
@@ -1,8 +0,0 @@
-declare global {
-    const __DEV__: boolean | undefined;
-}
-declare const _default: typeof globalThis & {
-    __DEV__: typeof __DEV__;
-};
-export default _default;
-//# sourceMappingURL=global.d.ts.map
\ No newline at end of file
diff --git a/node_modules/@apollo/client/utilities/globals/index.d.ts b/node_modules/@apollo/client/utilities/globals/index.d.ts
index b1ff5e3..5bd6ef5 100644
--- a/node_modules/@apollo/client/utilities/globals/index.d.ts
+++ b/node_modules/@apollo/client/utilities/globals/index.d.ts
@@ -3,6 +3,5 @@ import DEV from "./DEV";
 export { DEV };
 export declare function checkDEV(): void;
 export { maybe } from "./maybe";
-export { default as global } from "./global";
 export { invariant, InvariantError };
 //# sourceMappingURL=index.d.ts.map
\ No newline at end of file
```

**Comment by benjamn** [comment]
@danilobuerger Does it work if you replace the declaration with exactly the same type that React Native uses, or is the only option to remove the declarations?

**Comment by danilobuerger** [comment]
@benjamn I tried that at first, but getting the same error as above.

**Comment by benjamn** [comment]
@szymn @danilobuerger @mitsuhiko Please have a look at PR #9386 if you have a chance! I'll also release those changes in the next v3.6 beta release so you can try them out in your real applications more easily.

**Comment by benjamn** [comment]
These changes are now available from npm in `@apollo/client@3.5.9` (just published). Please give that version a try when you have a chance @szymn @danilobuerger @mitsuhiko!

**Comment by danilobuerger** [comment]
@benjamn can confirm that 3.5.9 no longer produces the error. Thanks!
